`zipfile.BadZipFile: File is not a zip file` when loading an `.xlsx` file

Not sure if this is a misconfiguration on my side, but I get the error: `zipfile.BadZipFile: File is not a zip file` both from reading an `.xlsx` from local/S3, or a pytest testcase saving and reloading an `openpyxl` created workbook.

## Debugging/Workaround

My current workaround is to modify: https://github.com/ets/tap-spreadsheets-anywhere/blob/main/tap_spreadsheets_anywhere/excel_handler.py#L64-L65 from:

```python
def get_row_iterator(table_spec, file_handle):
    workbook = openpyxl.load_workbook(file_handle, read_only=True)
```

to:

```python
def get_row_iterator(table_spec, file_handle):
    workbook = openpyxl.load_workbook(file_handle.name, read_only=True)
```

## Reproduction:
- Created an example testcase:
  ```python
  from openpyxl import Workbook
  from tap_spreadsheets_anywhere.format_handler import get_row_iterator


  def get_worksheet():
      """Create a basic workbook that can be manipulated for tests.
      See: https://openpyxl.readthedocs.io/en/stable/usage.html.
      """
      wb = Workbook()
      ws = wb.active
      tree_data = [
          ["Type", "Leaf Color", "Height"],
          ["Maple", "Red", 549],
          ["Oak", "Green", 783],
          ["Pine", "Green", 1204]
      ]
      exp_tree_data = [
          {'type': 'Maple', 'leaf_color': 'Red', 'height': 549},
          {'type': 'Oak', 'leaf_color': 'Green', 'height': 783},
          {'type': 'Pine', 'leaf_color': 'Green', 'height': 1204},
      ]
      [ws.append(row) for row in tree_data]
      return ws, wb, tree_data, exp_tree_data


  class TestFormatHandlerExcelSkipInitial:
      """pytests to validate Skip Initial for Excel files works as expected."""
      def test_parse_data(self, tmpdir):
          xlsx = tmpdir / "fake_test.xlsx"
          uri = f"file://{xlsx}"
          worksheet, workbook, _, exp = get_worksheet()
          workbook.save(xlsx)

          iterator = get_row_iterator({"format": "excel"}, uri)
          assert next(iterator) == exp[0]
  ```
- Current output:
  ```python
  cd ~/github_forks/tap-spreadsheets-anywhere/ && source .venv/bin/activate && pytest -vvvv -p no:warnings -k TestFormatHandlerExcelSkipInitial
  ============================================= test session starts ==============================================
  platform darwin -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0 -- /Users/craigastill/github_forks/tap-spreadsheets-anywhere/.venv/bin/python3.11
  cachedir: .pytest_cache
  rootdir: /Users/craigastill/github_forks/tap-spreadsheets-anywhere
  collected 30 items / 29 deselected / 1 selected                                                                

  tap_spreadsheets_anywhere/test/test_format.py::TestFormatHandlerExcelSkipInitial::test_parse_data FAILED [100%]

  =================================================== FAILURES ===================================================
  ______________________________ TestFormatHandlerExcelSkipInitial.test_parse_data _______________________________

  self = <tap_spreadsheets_anywhere.test.test_format.TestFormatHandlerExcelSkipInitial object at 0x1055e9e50>
  tmpdir = local('/private/var/folders/yx/p09k0hh165scp5p_ntthr29c0000gn/T/pytest-of-craigastill/pytest-6/test_parse_data0')

      def test_parse_data(self, tmpdir):
          xlsx = tmpdir / "fake_test.xlsx"
          uri = f"file://{xlsx}"
          worksheet, workbook, _, exp = get_worksheet()
          workbook.save(xlsx)

  >       iterator = get_row_iterator({"format": "excel"}, uri)

  tap_spreadsheets_anywhere/test/test_format.py:218: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  tap_spreadsheets_anywhere/format_handler.py:164: in get_row_iterator
      iterator = tap_spreadsheets_anywhere.excel_handler.get_row_iterator(table_spec, reader)
  tap_spreadsheets_anywhere/excel_handler.py:66: in get_row_iterator
      workbook = openpyxl.load_workbook(file_handle, read_only=True)
  .venv/lib/python3.11/site-packages/openpyxl/reader/excel.py:344: in load_workbook
      reader = ExcelReader(filename, read_only, keep_vba,
  .venv/lib/python3.11/site-packages/openpyxl/reader/excel.py:123: in __init__
      self.archive = _validate_archive(fn)
  .venv/lib/python3.11/site-packages/openpyxl/reader/excel.py:95: in _validate_archive
      archive = ZipFile(filename, 'r')
  /usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/zipfile.py:1301: in __init__
      self._RealGetContents()
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

  self = <zipfile.ZipFile [closed]>

      def _RealGetContents(self):
          """Read in the table of contents for the ZIP file."""
          fp = self.fp
          try:
              endrec = _EndRecData(fp)
          except OSError:
              raise BadZipFile("File is not a zip file")
          if not endrec:
  >           raise BadZipFile("File is not a zip file")
  E           zipfile.BadZipFile: File is not a zip file

  /usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/zipfile.py:1368: BadZipFile
  --------------------------------------------- Captured stdout call ---------------------------------------------
  <_io.TextIOWrapper name='/private/var/folders/yx/p09k0hh165scp5p_ntthr29c0000gn/T/pytest-of-craigastill/pytest-6/test_parse_data0/fake_test.xlsx' mode='r' encoding='utf-8'>
  =========================================== short test summary info ============================================
  FAILED tap_spreadsheets_anywhere/test/test_format.py::TestFormatHandlerExcelSkipInitial::test_parse_data - zi...
  ======================================= 1 failed, 29 deselected in 1.22s =======================================
  ```

## Workaround:

Adding the above mentioned workaround results in success, but not tested this in anger.

```python
cd ~/github_forks/tap-spreadsheets-anywhere/ && source .venv/bin/activate && pytest -vvvv -p no:warnings -k TestFormatHandlerExcelSkipInitial
============================= test session starts ==============================
platform darwin -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0 -- /Users/craigastill/github_forks/tap-spreadsheets-anywhere/.venv/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/craigastill/github_forks/tap-spreadsheets-anywhere
collected 30 items / 29 deselected / 1 selected                                

tap_spreadsheets_anywhere/test/test_format.py::TestFormatHandlerExcelSkipInitial::test_parse_data PASSED [100%]

====================== 1 passed, 29 deselected in 23.37s =======================
```

--- a/tap_spreadsheets_anywhere/excel_handler.py
+++ b/tap_spreadsheets_anywhere/excel_handler.py
@@ -62,7 +62,7 @@ def get_legacy_row_iterator(table_spec, file_handle):
 
 
 def get_row_iterator(table_spec, file_handle):
-    workbook = openpyxl.load_workbook(file_handle, read_only=True)
+    workbook = openpyxl.load_workbook(file_handle.name, read_only=True)
     
     if "worksheet_name" in table_spec:
         try:

